specification/security/_types/Privileges.ts

```diff
@@ -94,14 +94,27 @@ export class IndicesPrivileges {
   /**
    * A search query that defines the documents the owners of the role have read access to. A document within the specified indices must match this query for it to be accessible by the owners of the role.
    */
-  query?: string[] | QueryContainer | RoleTemplateQueryContainer
+  query?: IndicesPrivilegesQuery
   /**
    * Set to `true` if using wildcard or regular expressions for patterns that cover restricted indices. Implicitly, restricted indices have limited privileges that can cause pattern tests to fail. If restricted indices are explicitly included in the `names` list, Elasticsearch checks privileges against these indices regardless of the value set for `allow_restricted_indices`.
    * @server_default false
    */
   allow_restricted_indices?: boolean
 }
 
+/**
+ * While creating or updating a role you can provide either a JSON structure or a string to the API.
+ * However, the response provided by Elasticsearch will only be string with a json-as-text content.
+ *
+ * Since this is embedded in IndicesPrivileges, the same structure is used for clarity in both contexts.
+ *
+ * @codegen_names text, query, template
```

Comment:
Wondering if we should use `json_text` instead of just `text` so that the expected format of this string is more explicit.